aks_clusters.tf

```
@@ -20,7 +20,7 @@ module "aks_clusters" {
   admin_group_object_ids = try(each.value.admin_groups.azuread_group_keys, null) == null ? null : try(
     each.value.admin_groups.ids,
     [
-      for group_key in try(each.value.admin_groups.azuread_groups.keys, {}) : local.combined_objects_azuread_groups[local.client_config.landingzone_key][group_key].id
+      for group_key in try(each.value.admin_groups.azuread_groups_keys, {}) : local.combined_objects_azuread_groups[local.client_config.landingzone_key][group_key].id
```

@arnaudlh 
Not sure, if the "for...in..." expression is actually working, without the try/{} block, it fails:

```
  admin_group_object_ids = try(each.value.admin_groups.azuread_group_keys, null) == null ? null : try(
    each.value.admin_groups.ids,
    [
      for group_key in each.value.admin_groups.azuread_groups_keys : local.combined_objects_azuread_groups[local.client_config.landingzone_key][group_key].id
    ]
  )
```
Should be something like this:
```
  # admin_group_object_ids = can(each.value.admin_groups) == false ? null : try(
  #   each.value.admin_groups.ids,
  #   [
  #     for group_key in try(each.value.admin_groups.azuread_groups_keys,each.value.admin_groups.azuread_groups.keys) : local.combined_objects_azuread_groups[try(each.value.admin_groups.azuread_groups.lz_key, local.client_config.landingzone_key][group_key].id
  #   ]
  # )
```
But it should work if we define the block as:
```
    admin_groups = {
      # ids = ["12345c2e-6500-4684-b734-dd01c2dbc00d"] #
      azuread_group_keys = ["aks_cluster_admins"]
      # azuread_groups = {
      #   keys = ["aks_cluster_admins"]
      #  lz_key = "another-landingzone"
      # }
    }
```

The for...in expression should be evaluated.

---

@LaurentLesle any opinion on this one?

---

@LaurentLesle any update on this one please?

---

Let's keep: azuread_group.keys (keys part of azuread_group). easier to set the lz_key (remote groups) and the ids (by object_ids)

Let me know if that code is working and will approve the PR
```suggestion
      for group_key in try(each.value.admin_groups.azuread_group.keys, {}) : 
      coalesce(      
        try( local.combined_objects_azuread_groups[each.value.admin_groups.lz_key][group_key].id, null ),
        try( local.combined_objects_azuread_groups[local.client_config.landingzone_key][group_key].id, null )
      )
```